### PR TITLE
Add option to support custom http client

### PR DESCRIPTION
**Pull request by fidel-ruiz**
- add WithHTTPClient as an option

### CODE REVIEW

**Comment by sandersjessica** [comment]
Looks good to me!

File: options.go [hunk 1]
```diff
@@ -28,3 +31,10 @@ func WithLogger(logger LeveledLoggerInterface) Option {
 		c.log = logger
 	}
 }
+
+// WithHTTPClient allows to set a custom http client
```
Comment (joshsimard) [inline options.go:35]: `WithHTTPClient is used to set a custom http client` would sound better. `allows to` isn't the best choice in my opinion

**Review by joshsimard — CHANGES_REQUESTED** [review]

**Review by ysurtayev-bold — APPROVED** [review]

**Review by BoldBrandonM — DISMISSED** [review]

**Review by joshsimard — APPROVED** [review]

**Review by ysurtayev-bold — DISMISSED** [review]

**Review by gordcurrie — APPROVED** [review]
Looks good to me

**Review by gcfresh — APPROVED** [review]
Looks good to me

**Review by gordcurrie — APPROVED** [review]
Approving again as not the checks weren't getting triggered and @fidel-ruiz had to `git commit --amend` to trigger them to run again.

**Pull request merged**